tests/test_app/test_plugins.py

```diff
@@ -0,0 +1,163 @@
+            return data.upper()
+
+    mock_extension_manager([create_extension(uppercase)])
+
+    appercase = get_app("uppercase")
+    assert appercase("hello") == "HELLO"
+    assert appercase.__doc__ in _make_apphelp_docstring(appercase.__class__)
+
+
+def test_Install_app_function(mock_extension_manager):
+    @define_app
+    def uppercase(data: str) -> str:
+        """Test function that converts a string to uppercase"""
+        return data.upper()
+
+    mock_extension_manager([create_extension(uppercase)])
+
+    appercase = get_app("uppercase")
+    assert appercase("hello") == "HELLO"
+    assert appercase.__doc__ in _make_apphelp_docstring(appercase.__class__)
+
+
+@pytest.mark.parametrize("app_doc", [None, "text"])
+@pytest.mark.parametrize("init_doc", [None, "text"])
+def test_app_docs(mock_extension_manager, app_doc, init_doc):
+    @define_app
+    class documented_app:
+        """This is a test app that has a __init__, and a docstring"""
+
+        def __init__(self):
+            self.constant = 2
+
+        def main(self, val: int) -> int:
+            return val + self.constant
+
+    mock_extension_manager([create_extension(documented_app)])
+
+    assert cogent3.app.apps().names() == ["documented_app"]
+    app = get_app("documented_app")
+    app.__class__.__doc__ = app_doc
+    app.__class__.__init__.__doc__ = init_doc
+    app_help("documented_app")
+    got = _make_apphelp_docstring(app.__class__)
+    assert "Options" in got
+
+
+def test_namespace_collision(mock_extension_manager):
+    @define_app
+    class app1:
+        def main(self, data: str) -> str:
+            return data.upper()
+
+    @define_app
+    class app2:
+        def main(self, data: str) -> str:
+            return data.lower()
+
+    # create two apps with the same name and different modules
+    mock_extension_manager(
+        [
+            create_extension(app1, module_name="module1"),
+            create_extension(app2, name="app1", module_name="module2"),
+        ]
+    )
+
+    assert cogent3.app.apps().names() == ["app1", "app1"]
+
+    with pytest.raises(NameError):
+        _ = get_app(
+            "app1"
+        )  # request app by name only, when there are multiple apps with the same name
+
+    app_by_module_name_1 = get_app("module1.app1")  # request app by name and module
+    app_by_module_name_2 = get_app("module2.app1")
+    assert app_by_module_name_1("Hello") == "HELLO"
+    assert app_by_module_name_2("Hello") == "hello"
+
+    composition = app_by_module_name_1 + app_by_module_name_2
+    assert composition("Hello") == "hello"
+
+
+def test_available_apps_local(mock_extension_manager):
+    """available_apps robust to local scope apps"""
+
+    @define_app
+    def dummy(val: int) -> int:
+        return val
+
+    mock_extension_manager([create_extension(dummy)])
+    apps = available_apps()
+    assert isinstance(apps, Table)
+    apps.filtered(lambda x: dummy.__name__ == x, columns="name")

```

Comment:
this is having effect. Table's are immutable, so you'd need to be assign the result of the `filtered` call to `apps`